vm/src/builtins/tuple.rs

```diff
@@ -169,11 +169,17 @@ impl PyTuple {
 
     #[pymethod(name = "__mul__")]
     #[pymethod(name = "__rmul__")]
-    fn mul(&self, counter: isize, vm: &VirtualMachine) -> PyRef<Self> {
-        if self.elements.is_empty() || counter == 0 {
+    fn mul(zelf: PyRef<Self>, counter: isize, vm: &VirtualMachine) -> PyRef<Self> {
+        if zelf.elements.is_empty() || counter == 0 {
             vm.ctx.empty_tuple.clone()
+        } else if counter == 1 && zelf.clone_class().is(PyTuple::class(vm)) {
```

Comment:
No need to clone here:

```suggestion
        } else if counter == 1 && zelf.class().is(&vm.ctx.types.tuple_type) {
```